shared/src/ledger/eth_bridge/bridge_pool_vp.rs

```diff
@@ -116,11 +118,30 @@ where
         };
 
         let pending_key = get_pending_key(&transfer);
+        // check that transfer is not already in the pool
+        match (&self.ctx).read_pre(&pending_key) {
+            Ok(Some(_)) => {
+                tracing::debug!(
+                    "Rejecting transaction as the transfer is already in the \
+                     Ethereum bridge pool."
+                );
+                return Ok(false);
+            }
+            Err(e) => {
+                return Err(eyre!(
+                    "Could not read the storage key associated with the \
+                     transfer: {:?}",
+                    e
+                )
+                .into());
+            }
+            _ => {}
```

Comment:
Noticed this while reviewing https://github.com/anoma/namada/pull/715 but the change is coming from this PR I guess

We would match here only for the `Ok(None)` case which means the `pending_key` got deleted(?), I think we should return an `Ok(false)` here if someone tries to do this via a wasm transaction. We should be deleting pending keys from the pool in derived transactions only, once we receive a `TransfersToEthereum` event, if I understand correctly

Reply:
I replied to this comment on #715.

Reply:
Well, the reply disappeared. This means that the pending key was not used in storage. It may have been deleted (the transfer timed out), or not. We actually don't care.